javascripts/discourse/components/spreadsheet-editor.js

```diff
@@ -108,29 +108,46 @@ export default class SpreadsheetEditor extends GlimmerComponent {
     return this.buildSpreadsheet(data, columns);
   }
 
-  buildPopulatedTable(table) {
-    const tableObject = tableToObj(table);
-    const headings = [];
-    const tableData = [];
+  extractTableContent(data) {
+    return data
+      .flat()
+      .filter((t) => t.type === "inline")
+      .map((t) => t.content);
+  }
 
-    tableObject.forEach((object) => {
-      // Build Headings
-      if (!headings.includes(...Object.keys(object))) {
-        headings.push(...Object.keys(object));
+  buildPopulatedTable(tableTokens) {
+    const contentRows = tokenRange(tableTokens, "tr_open", "tr_close");
+    const rows = [];
+    let headings;
+    const rowWidthFactor = 8;
+
+    contentRows.forEach((row, index) => {
+      if (index === 0) {
+        // headings
+        headings = this.extractTableContent(row).map((heading) => {
```

Comment:
This is ok for now, but can also refactor to use thead as an identifier for the table header. Not sure how a table without headings would be handled currently.

Reply:
Ah yes, I haven't tried a table without headings yet. I can test that next.

I was previously using the `<th>` to check, but I saw the headings appear when I was looking at the rows, so I thought I'd just compile it into one.